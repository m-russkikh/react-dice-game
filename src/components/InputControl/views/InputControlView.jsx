import React from 'react';
import {
  FormGroup,
  Label,
  Col,
  Input,
} from 'reactstrap';

import { InputControlViewPropType } from '../propTypes';


function InputControlView({
  label, value, min, max,
}) {
  return (
    <FormGroup row>
      <Label sm="3" md="2">{label}</Label>
      <Col>
        <Input type="number" min={min} max={max} value={value} />
      </Col>
    </FormGroup>
  );
}

InputControlView.propTypes = InputControlViewPropType;

export default InputControlView;
