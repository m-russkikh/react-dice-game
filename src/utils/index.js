import { DEFAULT_BALANCE } from '../constants/balance';

export function getPreloadedState() {
  let balance = {
    value: DEFAULT_BALANCE,
  };

  try {
    const serializedBalance = localStorage.getItem('balance');

    if (serializedBalance) {
      balance = JSON.parse(serializedBalance);
    }
  } catch (err) {
  //  ignore errors
  }

  return { balance };
}

export function saveState(state) {
  const serializedBalance = JSON.stringify(state.balance);

  localStorage.setItem('balance', serializedBalance);
}

export function getNextBetNumber() {
  return Math.floor(Math.random(0, 1) * 100) + 1;
}
