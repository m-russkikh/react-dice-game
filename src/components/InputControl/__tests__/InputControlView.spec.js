import { getShallowWrapper } from '../../../utils/testUtils';
import InputControlView from '../views/InputControlView';


describe('InputControlView', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      label: 'label',
      onChange: jest.fn(),
    };

    wrapper = getShallowWrapper(InputControlView, props);
  });

  describe('render', () => {
    it('should display correct', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });

  describe('Input', () => {
    let input;

    beforeEach(() => {
      input = wrapper.find('Input');
    });

    describe('change event', () => {
      it('should handle change event', () => {
        const event = { currentTarget: { value: 5 } };

        input.simulate('change', event);

        expect(props.onChange).toHaveBeenCalledTimes(1);
        expect(props.onChange).toHaveBeenCalledWith(event);
      });
    });

    describe('keypress event', () => {
      it('should change input value', () => {
        const event = {
          key: 5,
          preventDefault: jest.fn(),
        };

        input.simulate('keypress', event);

        expect(event.preventDefault).toHaveBeenCalledTimes(0);
      });

      it('should not change input value', () => {
        const event = {
          key: 'r',
          preventDefault: jest.fn(),
        };

        input.simulate('keypress', event);

        expect(event.preventDefault).toHaveBeenCalledTimes(1);
      });
    });
  });
});
