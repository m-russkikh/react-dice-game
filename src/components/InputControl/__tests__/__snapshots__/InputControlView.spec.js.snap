// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputControlView render should display correct 1`] = `
"<FormGroup row={true} tag=\\"div\\">
  <Label sm=\\"3\\" md=\\"2\\" tag=\\"label\\" widths={{...}}>
    label
  </Label>
  <Col tag=\\"div\\" widths={{...}}>
    <Input onChange={[Function: mockConstructor]} onKeyPress={[Function: handleKeyPress]} type=\\"text\\" />
  </Col>
</FormGroup>"
`;
